new-version.sh

```diff
@@ -15,68 +15,74 @@
 
 set -e
 
-# PYTHON
+usage() {
+  echo "usage: ./$(basename -- ${0}) <NEW_VERSION> <NEXT_VERSION>"
+  exit 1
+}
+
+readonly SEMVER_REGEX="^[0-9]+(\.[0-9]+){2}(-rc\.[0-9]+)?(-SNAPSHOT)?$" # X.Y.Z
+                                                                        # X.Y.Z-rc.*
+                                                                        # X.Y.Z-SNAPSHOT
+
+# Change working directory to project root
+project_root=$(git rev-parse --show-toplevel)
+cd "${project_root}"
 
 # Verify bump2version is installed
 if [[ ! $(type -P bump2version) ]]; then
- echo "bump2version not installed! Please see https://github.com/c4urself/bump2version#installation"
- exit 1
+  echo "bump2version not installed! Please see https://github.com/c4urself/bump2version#installation"
+  exit 1
 fi
 
-if [[ -z "${type}" ]]; then
-  # Default to 'patch'
-  type="patch"
+branch=$(git symbolic-ref --short HEAD)
+if [[ "${branch}" != "main" ]]; then
+  echo "Error: You may only release on 'main'!"
+  exit 1;
 fi
 
-
-# get new version tag from CI
-if [[ -n "$1" ]]; then
-  # Default to current tag
-  NEW_VERSION=$1
-elif [[ -n ${CIRCLE_TAG} ]]; then
-  NEW_VERSION=${CIRCLE_TAG}
-else
-  echo "Version not passed to script. Use ./new-version.sh VERSION"
-  exit 1
+if [[ $# -eq 0 ]] ; then
+  usage
 fi
 
-# Bump marquez_client version
-VERSION=$(python ./clients/python/setup.py --version)
-bump2version \
-  --config-file ./clients/python/setup.cfg \
-  --current-version "${VERSION}" \
-  --new-version "${NEW_VERSION}" \
-  --no-commit \
-  --no-tag \
-  --allow-dirty \
-  "${type}" ./clients/python/marquez_client/version.py
-
-
-# Bump marquez_airflow version
-VERSION=$(python ./integrations/airflow/setup.py --version)
-bump2version \
-  --config-file ./integrations/airflow/setup.cfg \
-  --current-version "${VERSION}" \
-  --new-version "${NEW_VERSION}" \
-  --no-commit \
-  --no-tag \
-  --allow-dirty \
-  "${type}" ./integrations/airflow/marquez_airflow/version.py
-
-
-GRADLE_PROPS=(
-  gradle.properties
-  api/gradle.properties
-  integrations/spark/gradle.properties
-)
-
-for FILE in "${GRADLE_PROPS[@]}"
-do
-  sed -i 's/\(^version=.*$\)/version='"${NEW_VERSION}"'/g' "$FILE"
+NEW_VERSION="${1}"
+NEXT_VERSION="${2}"
+
+# Ensure valid versions
+VERSIONS=($NEW_VERSION $NEXT_VERSION)
+for VERSION in "${VERSIONS[@]}"; do
+  if [[ ! "${VERSION}" =~ ${SEMVER_REGEX} ]]; then
+    echo "Error: Version '${VERSION}' must match '${SEMVER_REGEX}'"
+    exit 1
+  fi
 done
 
+# (1) Bump python module versions
+PYTHON_MODULES=(clients/python/ integrations/airflow/)
+for PYTHON_MODULE in "${PYTHON_MODULES[@]}"; do
+  (cd "${PYTHON_MODULE}" && bump2version manual --new-version "${NEW_VERSION}")
+done
+
+# (2) Bump java module versions
+sed -i "" "s/version=.*/version=${NEW_VERSION}/g" gradle.properties
+
+# (3) Prepare release commit
+git commit -am "Prepare for release ${NEW_VERSION}"
+
+# (4) Prepare release tag
+git fetch --all --tags > /dev/null 2>&1
```

Comment:
This command has failed for me in the past if the local tags are desynchronized from the remote server. I would recommend adding a `-f` (to force the fetch) or at least catching the error here.

Reply:
I'll catch the error 👍